### PR TITLE
Upgrade bundled Postgres to 11.1

### PR DESCRIPTION
**Pull request by tsenart**
This commit upgrades the version of Postgres we run in development and in new Sourcegraph deployments with the server image, in time for the 3.0-beta release. Existing deployments need to undergo an upgrade procedure which I'll be working on testing and documenting this week.

Issue: #1404

Related:
- https://github.com/sourcegraph/deploy-sourcegraph/pull/161
- https://github.com/sourcegraph/infrastructure/pull/993

### CODE REVIEW

**Comment by tsenart** [comment]
@sourcegraph/team: You'll need to go through the README again to [re-initialize the database](https://github.com/sourcegraph/sourcegraph/blob/master/doc/dev/local_development.md#step-4-initialize-your-database).

Make sure you have no PGXXX env vars set when doing so.

File: CHANGELOG.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ All notable changes to Sourcegraph are documented in this file.
 
 ### Changed
 
+- The Postgres database backing Sourcegraph has been upgraded from 9.4 to 11.1. Existing Sourcegraph users must conduct an [upgrade procedure](https://docs.sourcegraph.com/admin/postgres_9_4_to_11_1_upgrade)
```
Comment (keegancsmith) [inline CHANGELOG.md]: should we maybe have a single document called postgres_upgrade with an anchor to 9.4 to 11.1. Alt hough in future we hopefully have automated upgrades.

File: cmd/server/dockerfile.go [hunk 2]
```diff
@@ -12,14 +12,13 @@ package main
...
+// At the time of the 3.0-beta release, we ran Postgres 9.4 in sourcegraph.com
+// and existing customer deployments. With this release, we're upgrading to
+// 11.1 for new users and will follow up with providing documentation and
+// automation for existing deployments to be upgraded safely.
+// See: https://github.com/sourcegraph/sourcegraph/issues/1404
+
+//docker:repository edge
+//docker:install 'postgresql=11.1-r0' 'postgresql-contrib=11.1-r0' su-exec
 
 //docker:install redis
```
Comment (keegancsmith) [inline cmd/server/dockerfile.go]: I wonder if changing to edge updates the version of redis we are using? Are redis major versions upgrades backwards compatible with what is written to disk?

File: cmd/server/shared/shared_test.go [hunk 3]
```diff
@@ -1,35 +0,0 @@
...
-				switch parts[0] {
-				case "install":
-					install = append(install, strings.Fields(parts[1])...)
-				}
-			}
-		}
-	}
-
-	for _, pkg := range install {
-		if pkg == "'postgresql<9.7'" {
```
Comment (keegancsmith) [inline cmd/server/shared/shared_test.go]: we should keep this test to ensure people keep specifying exact versions, just to prevent accidental upgrades.

Comment (tsenart) [reply cmd/server/shared/shared_test.go]: 👍

File: cmd/server/dockerfile.go (hunk 2 above)
Comment (tsenart) [reply cmd/server/dockerfile.go]: Good catch. I'll check.

File: CHANGELOG.md (hunk 1 above)
Comment (tsenart) [reply CHANGELOG.md]: Yes, that's better.

**Comment by tsenart** [comment]
> What is your plan around upgrading? Using those unofficial (but defacto) upgrade docker images?

Upgrading what exactly? Customer deployments? Sourcegraph.com? Both of those are still to be done.

File: cmd/server/dockerfile.go [hunk 4]
```diff
@@ -12,14 +12,13 @@ package main
...
-// We run 9.4 in production, but if we are embedding might as well get
-// something modern, 9.6. We add the version specifier to prevent accidentally
-// upgrading to an even newer version.
-// NOTE: We have to stay at 9.6, otherwise existing users databases won't run
-// due to needing to be upgraded. There is no nice auto-upgrade we have here
-// without some engineering investment.
-
-//docker:repository v3.6
-//docker:install 'postgresql<9.7' 'postgresql-contrib<9.7' su-exec
+// At the time of the 3.0-beta release, we ran Postgres 9.4 in sourcegraph.com
```
Comment (nicksnyder) [inline cmd/server/dockerfile.go]:
```suggestion
// Prior to the 3.0-beta release, we ran Postgres 9.4 in sourcegraph.com
```

File: cmd/server/dockerfile.go [hunk 5]
```diff
@@ -12,14 +12,13 @@ package main
...
+// and existing customer deployments. With this release, we're upgrading to
```
Comment (nicksnyder) [inline cmd/server/dockerfile.go]:
```suggestion
// and existing customer deployments. With the 3.0-beta release, we're upgrading to
```

**Comment by tsenart** [comment]
> Customer deployments. Note when you merge this it will ship to dogfood => you need to atleast have a plan for server otherwise our dogfood instance will go down.

Ouch. I didn't plan for that. I'll see what I can do shortly.

**Pull request merged**